site/docs/expressions/specialized_record_expressions.md

```diff
@@ -68,10 +68,15 @@ Multi Value:
 For single value expressions, these are a compact equivalent of `expression = value1 OR expression = value2 OR .. OR expression = valueN`. When using an expression of this type, two things are required; the types of the test expression and all value expressions that are related must be of the same type. Additionally, a function signature for equality must be available for the expression type used.
 
 
+## Coalesce Expression
 
+Coalesce expression evaluates arguments from left to right and return the first argument encountered that is not null. Once a non-null argument is found, the remaining arguments are not evaluated. If all arguments are null, return null.
 
+The arguments could be both nullable and non-nullable. If there is any argument that is non-nullable, the return type would be non-nullable. If all the arguments are nullable, or there is no argument, the return type would be nullable.
```

Comment:
I'd add something stating that, nullability aside, the types must match, and that at least one argument is required (our null value is typed, so there is no generic null for an empty coalesce() expression to return). I'd also use less passive language in general, but that's more of a nit.

```suggestion
The arguments may be any combination of nullable and non-nullable, but must otherwise be of the same type. If any argument is non-nullable, the return type is non-nullable. If all the arguments are nullable, or there is no argument, the return type is be nullable. Nullability aside, the return type matches the type of the arguments. At least one argument is required, in order for the return type to be defined.
```